File: chat-uikit/src/main/java/io/agora/chat/uikit/chat/interfaces/OnReactionMessageListener.java
package io.agora.chat.uikit.chat.interfaces;


import io.agora.chat.ChatMessage;

public interface OnReactionMessageListener {

    /**
     * add reaction success
     *
     * @param message
     */
    void addReactionMessageSuccess(ChatMessage message);

    /**
     * add reaction fail
     *
     * @param message
     * @param code
     * @param error
     */
    void addReactionMessageFail(ChatMessage message, int code, String error);

    /**
     * remove reaction success
     *
     * @param message
     */
    void removeReactionMessageSuccess(ChatMessage message);

    /**
     * remove reaction fail
     *
     * @param message
     * @param code
     * @param error
     */
    void removeReactionMessageFail(ChatMessage message, int code, String error);

}

File: ease-im-kit/src/main/kotlin/io/agora/chat/uikit/feature/chat/interfaces/OnReactionMessageListener.kt
package io.agora.chat.uikit.feature.chat.interfaces

import io.agora.chat.uikit.common.ChatMessage


interface OnReactionMessageListener {
    /**
     * add reaction success
     *
     * @param message
     */
    fun addReactionMessageSuccess(message: ChatMessage?)

    /**
     * add reaction fail
     *
     * @param message
     * @param code
     * @param error
     */
    fun addReactionMessageFail(message: ChatMessage?, code: Int, error: String?)

    /**
     * remove reaction success
     *
     * @param message
     */
    fun removeReactionMessageSuccess(message: ChatMessage?)

    /**
     * remove reaction fail
     *
     * @param message
     * @param code
     * @param error
     */
    fun removeReactionMessageFail(message: ChatMessage?, code: Int, error: String?)
}